src/util.js

```diff
@@ -276,6 +276,51 @@ function copyDescriptors(from, to, filter) {
     }
 }
 
+// This function should not be used for methods that return a value
+// Using this function implies small performance overhead, should be used only
+// for debug methods
+function hookTo(prototypeObject, methodName, extension) {
+    var existingMethodImpl = prototypeObject[methodName];
+    if (typeof existingMethodImpl === "function" &&
+        existingMethodImpl.extensions) {
+        var extensions = existingMethodImpl.extensions;
+        extensions[extension.toString()] = extension;
+        // Performance optimization: remove noops
```

Comment:
I understand that you need the noop hook-methods, but I don't see any reason to place empty functions in these `extension` collections.
Just call `util.hookTo(Promise.prototype, "_promise…", null);` instead of `util.hookTo(Promise.prototype, "_promise…", function() {});` in the setup above, and initialise the `methodImpl` but ignore non-function values when adding to `extensions`.